DatabaseService/docker_context/handlers/PostgresHandler.py

```diff
@@ -134,6 +141,114 @@ async def handleGetUserStocksCommand(self, user_id, stock_id):
+    async def handleGetAllBuyTriggers(self):
+        get_triggers_query = from_table_where_query.format(
+            columns='*',
+            table=buy_triggers_table,
+            where=''
+        )
+
+        result = await self.fetchQuery(get_triggers_query)
+
+        if result is None:
+            return {
+                'errorMessage': 'Unexpected error. Could not get buy triggers.'
+            }, 404
+
+        resp_list = {}
+        for record in result:
+            stocks_list = {}
+            if record[0] in resp_list:
+                stocks_list = resp_list[record[0]]
+
+            stocks_list[record[1]] = [record[2], record[3]]
+            resp_list[record[0]] = stocks_list
+
+        return resp_list, 200
+        
+    async def handleGetUserBuyTriggers(self, user_id):
+        user_exists = await self._checkUserExists(user_id)
+
+        if not user_exists:
+            return {
+                'status': 'failure', 'message': 'Specified user does not exist.'
+            }, 404
+        
+        get_triggers_query = from_table_where_user_query.format(
+            columns='*',
+            table=buy_triggers_table,
+            user=user_id
+        )
+
+        result = await self.fetchQuery(get_triggers_query)
+
+        if result == []:
+            return {'errorMessage':'Triggers not found.'}, 404
+        elif result is None:
+            return {
+                'errorMessage': 'Unexpected error. Could not get triggers.'
+            }, 404
+
+        resp_list = {}
+        for record in result:
+            resp_list[record[1]] = [record[2], record[3]]
+
+        return resp_list, 200
+
+    async def handleGetAllSellTriggers(self):
+        get_triggers_query = from_table_where_query.format(
+            columns='*',
+            table=sell_triggers_table,
+            where=''
+        )
+
+        result = await self.fetchQuery(get_triggers_query)
+
+        if result is None:
+            return {
+                'errorMessage': 'Unexpected error. Could not get sell triggers.'
+            }, 404
+
+        resp_list = {}
+        for record in result:
+            stocks_list = {}
+            if record[0] in resp_list:
+                stocks_list = resp_list[record[0]]
+
+            stocks_list[record[1]] = [record[2], record[3]]
+            resp_list[record[0]] = stocks_list
+
+        return resp_list, 200
+
+    async def handleGetUserSellTriggers(self, user_id):
+        user_exists = await self._checkUserExists(user_id)
+
+        if not user_exists:
+            return {
+                'status': 'failure', 'message': 'Specified user does not exist.'
+            }, 404
+
+        get_triggers_query = from_table_where_user_query.format(
+            columns='*',
+            table=sell_triggers_table,
+            user=user_id
+        )
+
+        result = await self.fetchQuery(get_triggers_query)
+
+        if result == []:
+            return {'errorMessage':'Triggers not found.'}, 404
```

Comment:
would this necessarily be an error or just that the user does not have any triggers set

Reply:
Good point actually. This also means that I should change other user-specific GET endpoints to not throw an error in this case, as they currently are.

Reply:
Will handle in follow up PR.